graph/resolver/schema.resolvers.go

```diff
@@ -45,63 +92,154 @@ func (r *mutationResolver) CreateDiscussion(ctx context.Context, anonymityType m
-		}
-		creatingUser.User = user
+	flair, err := r.DAOManager.GetFlairByID(ctx, flairID)
+	if err != nil {
+		return nil, err
+	} else if flair == nil {
+		return nil, fmt.Errorf("Error fetching flair with ID (%s)", flairID)
 	}
 
-	discussion, err := r.DAOManager.GetDiscussionByID(ctx, discussionID)
-	if discussion == nil || err != nil {
-		return nil, fmt.Errorf("Discussion with ID %s not found", discussionID)
+	participant, err := r.DAOManager.GetParticipantByID(ctx, participantID)
+	if err != nil {
+		return nil, err
+	} else if participant == nil {
+		return nil, fmt.Errorf("Error fetching participant with ID (%s)", participantID)
+	}
+
+	if *participant.UserID != flair.UserID {
+		// Flair does not belong to this participant, bad request.
+		// Note: This error message is intentionally ambiguous as to not reveal
+		// the lack of flair for a participant.
+		return nil, fmt.Errorf("Unathorized")
+	}
+
+	// TODO: Currently this only allows the requesting user to assign flair to
+	// their own participant. But this API takes in userID so that an admin or
+	// moderator could potentially assign flair on behalf of another
+	// participant. To add this functionality remove the following line and
+	// add authorization/permission logic.
+	if currentUser.UserID != *participant.UserID {
+		return nil, fmt.Errorf("Unauthorized")
+	}
+
+	return r.DAOManager.AssignFlair(ctx, *participant, flairID)
+}
+
+func (r *mutationResolver) UnassignFlair(ctx context.Context, participantID string) (*model.Participant, error) {
+	currentUser := auth.GetAuthedUser(ctx)
+	if currentUser == nil {
+		return nil, fmt.Errorf("Need auth")
 	}
 
-	participants, err := r.DAOManager.GetParticipantsByDiscussionID(ctx, discussionID)
+	participant, err := r.DAOManager.GetParticipantByID(ctx, participantID)
 	if err != nil {
 		return nil, err
+	} else if participant == nil {
+		return nil, fmt.Errorf("Error fetching participant with ID (%s)", participantID)
 	}
-	var participant *model.Participant
-	for _, p := range participants {
-		if p.UserID != nil && *p.UserID == creatingUser.UserID {
-			participant = &p
-			break
-		}
+
+	// TODO: Currently this only allows the requesting user to unassign flair
+	// from their own participant. But this API takes in userID so that an admin
+	// or moderator could potentially unassign flair on behalf of another
+	// participant. To add this functionality remove the following line and
+	// add authorization/permission logic.
+	if currentUser.UserID != *participant.UserID {
+		return nil, fmt.Errorf("Unauthorized")
 	}
-	if participant == nil {
-		return nil, fmt.Errorf("Current user not a participant in this discussion")
+
+	return r.DAOManager.UnassignFlair(ctx, *participant)
+}
+
+func (r *mutationResolver) CreateFlairTemplate(ctx context.Context, displayName *string, imageURL *string, source string) (*model.FlairTemplate, error) {
+	currentUser := auth.GetAuthedUser(ctx)
```

Comment:
Can you add a todo that only an Admin can do this, please? I know we don't have the concept of an admin yet but it would be good to track that.

Reply:
✅ https://github.com/nedrocks/delphisbe/pull/2/commits/316b8d3a22c3e31bb22050f14cbe6c5c42a9bf1a